schema/auth.json

```diff
@@ -0,0 +1,214 @@
+          "description": "Defines the auth type",
+          "enum": [
+            "basic",
+            "bearer",
+            "oidc"
+          ],
+          "default": "basic"
+        },
+        "properties": {
+          "oneOf": [
+            {
+              "type": "string",
+              "description": "Expression referencing a workflow secret that contains all needed auth info"
+            },
+            {
+              "title": "Basic Auth Info",
+              "$ref": "#/definitions/basicinfodef"
+            },
+            {
+              "title": "Bearer Auth Info State",
+              "$ref": "#/definitions/bearerinfodef"
+            },
+            {
+              "title": "Oidc Auth Info",
+              "$ref": "#/definitions/oidcinfodef"
+            }
+          ]
+        }
+      },
+      "required": [
+        "name",
+        "info",
+        "scheme"
+      ]
+    },
+    "basicinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed basic auth info"
+        },
+        {
+          "type": "object",
+          "description": "Basic auth information",
+          "properties": {
+            "username": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the user name",
+              "minLength": 1
+            },
+            "password": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the user password",
+              "minLength": 1
+            },
+            "metadata": {
+              "$ref": "common.json#/definitions/metadata"
+            }
+          },
+          "required": [
+            "username",
+            "password"
+          ],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "bearerinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed bearer auth info"
+        },
+        {
+          "type": "object",
+          "description": "Bearer auth information",
+          "properties": {
+            "token": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the token",
+              "minLength": 1
+            },
+            "metadata": {
+              "$ref": "common.json#/definitions/metadata"
+            }
+          },
+          "required": [
+            "token"
+          ],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "oidcinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed oidc auth info"
+        },
+        {
+          "type": "object",
+          "description": "Oidc auth information",
+          "properties": {
```

Comment:
`authority: https://keycloak.local`
The runtime then knows to contact https://keycloak.local/.well-known/openid-configuration to get the file that describe the openid endpoints

Reply:
added